kpatch-build/create-diff-object.c

```diff
@@ -2605,7 +2647,8 @@ void kpatch_create_mcount_sections(struct kpatch_elf *kelf)
 					list);
 		if (rela->type != R_X86_64_NONE ||
 		    strcmp(rela->sym->name, "__fentry__"))
-			ERROR("function '%s' has no fentry call; unable to patch", sym->name);
+			ERROR("%s: the first relocation is not for a fentry call",
+			      sym->name);
```

Comment:
I cannot say if it is possible to hit the error there. I kept it a kind of a sanity check, like the one for 'index' below:
  if (index != nr)
		ERROR("size mismatch in funcs sections");

If new symbol structures cannot appear in the list somehow after kpatch_find_fentry_calls() is called, then, I think, the check is not necessary.